autofit/messages/transformed.py

```diff
@@ -0,0 +1,171 @@
+import inspect
+from typing import Type, Tuple, Union
+
+import numpy as np
+
+from .abstract import AbstractMessage
+from .transform import AbstractDensityTransform
+from ..tools.cached_property import cached_property
+
+
+class TransformedMessage(AbstractMessage):
+    _Message: Type[AbstractMessage]
+    _transform: Union[AbstractDensityTransform, Type[AbstractDensityTransform]]
+    _depth = 0
+
+    def __init__(self, *args, **kwargs):
```

Comment:
It might be nice/better/easier to use `__new__` to create a new class with the _transform attributed initialised, but it's probably less work to fix this version of the implementation.

Reply:
You could also define two classes, 
```
class UninitialisedTransformedMessage(TransformedMessage):
     _Message: Type[AbstractMessage]
     _transform: Type[AbstractDensityTransform]
     
     def __new__(cls, *args, **kwargs):
           transform_args = inspect.getfullargspec(
                self._transform
            ).args[1:]

            transform_dict = dict()
            for arg in transform_args:
                if arg in kwargs:
                    transform_dict[
                        arg
                    ] = kwargs.pop(arg)
            transform = cls._transform(
                **transform_dict
            )
            return cls._Message.transformed(transform)
```

Reply:
Hmm that might be nicer. One of my concerns at the moment is the inconsistency with which I've converted classmethods. Some will just fail for bad reasons. Will consider trying this